### ISSUE TITLE
Duplicate does not have permission to a lot of data in macOS Mojave

### ISSUE THREAD

**Issue by iBobik** [duplicati/duplicati#3403]
- [x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.3.11 (2.0.3.11_canary_2018-09-05)
- **Operating system**: macOS Mojave
- **Backend**: WebDAV

## Description

After update to Mojave it outputs many FileAccessError errors to a lot of locations:
```
 [Warning-Duplicati.Library.Main.Operation.Backup.FileEnumerationProcess-FileAccessError]: Error reported while accessing file: /Users/Bobik/Pictures/Knihovna fotografií.photoslibrary/,
```
Locations in my case are: Photos libraries, Library/Suggestions and Library/Safari

This locations seems like that what are protected by system from Mojave - user needs to give permission to app to allow access there.

## Steps to reproduce

1. Upgrade to macOS Mojave
2. Run backup of entire Home directory
3. Wait until backup ends and prints strange errors

- **Actual result**:
Do not have access to a lot of files and print a lot of errors about it.
- **Expected result**:
Asks user for a permission to have full access to disc, so it will be able to backup all files like previous.

## Debug log

```
DeletedFiles: 2
DeletedFolders: 0
ModifiedFiles: 149
ExaminedFiles: 342024
OpenedFiles: 150
AddedFiles: 1
SizeOfModifiedFiles: 134376688
SizeOfAddedFiles: 460
SizeOfExaminedFiles: 44487850954
SizeOfOpenedFiles: 140113569
NotProcessedFiles: 0
AddedFolders: 0
TooLargeFiles: 0
FilesWithError: 0
ModifiedFolders: 0
ModifiedSymlinks: 0
AddedSymlinks: 0
DeletedSymlinks: 0
PartialBackup: False
Dryrun: False
MainOperation: Backup
CompactResults:
    DeletedFileCount: 0
    DownloadedFileCount: 0
    UploadedFileCount: 0
    DeletedFileSize: 0
    DownloadedFileSize: 0
    UploadedFileSize: 0
    Dryrun: False
    MainOperation: Compact
    ParsedResult: Warning
    Version: 2.0.3.11 (2.0.3.11_canary_2018-09-05)
    EndTime: 01.10.2018 3:01:17 (1538355677)
    BeginTime: 01.10.2018 3:00:43 (1538355643)
    Duration: 00:00:34.1594170
    Messages: [
        2018-10-01 02:50:32 +02 - [Information-Duplicati.Library.Main.Controller-StartingOperation]: Operace Backup zahájena,
        2018-10-01 02:53:43 +02 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Started:  (),
        2018-10-01 02:53:46 +02 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Completed:  (2,64 KB),
        2018-10-01 02:58:25 +02 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: Put - Started: duplicati-be7cd1125bef44140a6d71a9b86a14c02.dblock.zip.aes (2,06 MB),
        2018-10-01 02:59:20 +02 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: Put - Completed: duplicati-be7cd1125bef44140a6d71a9b86a14c02.dblock.zip.aes (2,06 MB),
...
    ]
    Warnings: [
        2018-10-01 02:50:43 +02 - [Warning-Duplicati.Library.Main.Operation.Backup.FileEnumerationProcess-FileAccessError]: Error reported while accessing file: /Users/Bobik/Pictures/Knihovna fotografií.photoslibrary/,
        2018-10-01 02:50:43 +02 - [Warning-Duplicati.Library.Main.Operation.Backup.FileEnumerationProcess-FileAccessError]: Error reported while accessing file: /Users/Bobik/Pictures/Knihovna fotografií.photoslibrary/,
        2018-10-01 02:50:44 +02 - [Warning-Duplicati.Library.Main.Operation.Backup.FileEnumerationProcess-FileAccessError]: Error reported while accessing file: /Users/Bobik/Library/Suggestions/,
        2018-10-01 02:50:44 +02 - [Warning-Duplicati.Library.Main.Operation.Backup.FileEnumerationProcess-FileAccessError]: Error reported while accessing file: /Users/Bobik/Library/Suggestions/,
        2018-10-01 02:50:44 +02 - [Warning-Duplicati.Library.Main.Operation.Backup.FileEnumerationProcess-FileAccessError]: Error reported while accessing file: /Users/Bobik/Library/Safari/,
...
    ]
... 
```

## Workaround
Open System settings -> Security and privacy -> Privacy -> Full access to disc and add Druplicati app there.

**Comment by Pectojin** [comment]
This was also discussed on the forum: https://forum.duplicati.com/t/file-access-on-macos-mojave/4735

I'm not sure we can do much beyond detecting it and putting a message in the UI for it.  It's an OS "limitation" imposed for security reasons.

I couldn't find any references of an API for checking if we have access, so it might have to be a pretty silly solution... https://forums.developer.apple.com/thread/107546

**Comment by kenkendk** [comment]
AFAIK, this is due to the new notary signing requirements in Mojave. 

The real fix is to write a new `info.plist` file that has permissions to those folders, and then update the build process to submit the `.dmg` and `.pkg` to the Apple notary service and then publish the notarised builds.

**Comment by sp-christian** [comment]
Hack: Grand "Full Disk Access" for `/bin/bash`.
See installation instruction: [Running on MacOS Catalina Beta](https://forum.duplicati.com/t/running-on-macos-catalina-beta/)

**Comment by kenkendk** [comment]
This was fixed in `2.0.5.0` (i.e. latest beta release).
